lib/Conversion/ExportVerilog/LegalizeNames.cpp

```diff
@@ -271,6 +278,160 @@ void GlobalNameResolver::gatherEnumPrefixes(mlir::ModuleOp topLevel) {
   }
 }
 
+void GlobalNameResolver::legalizeAnonEnums(mlir::ModuleOp topLevel) {
+  auto *context = topLevel->getContext();
+  auto loc = topLevel.getLoc();
+
+  // Creates a TypeScope on demand for anonymous enumerations.
+  TypeScopeOp typeScope;
+  auto getTypeScope = [&]() {
+    if (!typeScope) {
+      auto builder = OpBuilder::atBlockBegin(&topLevel.getRegion().front());
+      typeScope = builder.create<TypeScopeOp>(topLevel.getLoc(), "Enums");
+      typeScope.getBodyRegion().push_back(new Block());
+      mlir::SymbolTable symbolTable(topLevel);
+      symbolTable.insert(typeScope);
+    }
+    return typeScope;
+  };
+
+  // Helper to create TypeDecls and TypeAliases for EnumTypes;
+  unsigned enumCount = 0;
+  DenseMap<Type, Type> enumTypeAliases;
+  auto getEnumTypeDecl = [&](EnumType type) {
+    auto &typeAlias = enumTypeAliases[type];
+    if (typeAlias)
+      return typeAlias;
+    auto typeScope = getTypeScope();
+    auto builder = OpBuilder::atBlockEnd(&typeScope.getRegion().front());
+    auto declName = StringAttr::get(context, "enum" + Twine(enumCount++));
+    builder.create<TypedeclOp>(loc, declName, TypeAttr::get(type), nullptr);
+    auto symRef = SymbolRefAttr::get(typeScope.getSymNameAttr(),
+                                     FlatSymbolRefAttr::get(declName));
+    typeAlias = TypeAliasType::get(symRef, type);
+    return typeAlias;
+  };
+
+  // Process a type, replacing any anonymous enumerations with contained within.
+  std::function<Type(Type)> processType = [&](Type type) -> Type {
+    Type newType;
+    if (auto structType = dyn_cast<StructType>(type)) {
+      bool changed = false;
+      SmallVector<StructType::FieldInfo> fields;
+      for (auto &element : structType.getElements()) {
+        if (auto newFieldType = processType(element.type)) {
+          changed = true;
+          fields.push_back({element.name, newType});
```

Comment:
Err, what's "newType" supposed to be here? Haven't finished reading/grok'ing this loop but offhand looks like it may be used uninitialized/null here?

Reply:
oh, meant to be `newFieldType`?

Reply:
I deleted that "newType" local that I wasn't even using.. And yeah that is supposed to be the `newFieldType` not `newType` 🙈